### PR TITLE
[tests-only][full-ci]Store federated ocis user and token for e2e tests

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR update store for federated oCIS instance. On follow-up PR, tests related to oCIS federation will be added.

## Related Issue

- https://github.com/owncloud/web/issues/11680

## How Has This Been Tested?

- Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/e2e/cucumber/environment/index.ts [hunk 1]
```diff
@@ -136,12 +147,20 @@ After(async function (this: World, { result, willBeRetried }: ITestCaseHookParam
     await refreshAccessTokenForKeycloakOcisUser(user)
   }
 
-  await cleanUpUser(this.usersEnvironment.getUser({ key: 'admin' }))
+  await cleanUpUser(createdUserStore, this.usersEnvironment.getUser({ key: 'admin' }))
+
+  const tags = pickle.tags.map((tag) => tag.name)
+  if (tags.includes('@ocm')) {
```
Comment (saw-jan) [inline tests/e2e/cucumber/environment/index.ts:153]: Can we have a function to determine if current scenario is ocm or not

File: tests/e2e/cucumber/features/ocm/ocm.feature [hunk 2]
```diff
@@ -0,0 +1,20 @@
+@ocm
+Feature: federation management
+
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/ocm/ocm.feature:3]: webUI tests can have a user story

File: tests/e2e/cucumber/features/ocm/ocm.feature [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  Scenario: user create federated share
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/ocm/ocm.feature:4]:
```suggestion
  Scenario: user creates a federated share
```

File: tests/e2e/cucumber/steps/ui/session.ts [hunk 4]
```diff
@@ -118,3 +118,16 @@ When(
...
+Given('using {string} server', async function (this: World, server: string): Promise<void> {
+  switch (server) {
+    case 'LOCAL':
+      config.federatedServer = false
+      break
+    case 'FEDERATED':
+      config.federatedServer = true
+      break
+    default:
+      throw new Error('Server type out of scope')
```
Comment (saw-jan) [inline tests/e2e/cucumber/steps/ui/session.ts:131]:
```suggestion
      throw new Error(`Invalid server type: ${server}\nUse one of these: [LOCAL, FEDERATED]`)
```

File: tests/e2e/cucumber/environment/index.ts [hunk 5]
```diff
@@ -78,13 +83,18 @@ Before(async function (this: World, { pickle }: ITestCaseHookParameter) {
...
+  const tags = pickle.tags.map((tag) => tag.name)
   if (!config.basicAuth) {
     const user = this.usersEnvironment.getUser({ key: 'admin' })
     if (config.keycloak) {
       await setAccessTokenForKeycloakOcisUser(user)
       await setAccessTokenForKeycloakUser(user)
     } else {
       await setAccessAndRefreshToken(user)
+      if (tags.includes('@ocm')) {
+        config.federatedServer = true
```
Comment (saw-jan) [inline tests/e2e/cucumber/environment/index.ts:95]: only get token and let `using "LOCAL" server` step handle the `config.federatedServer`?
```suggestion
```

Comment (amrita-shrestha) [reply tests/e2e/cucumber/environment/index.ts:95]: if so how you plan to delete federatedoCIS

File: tests/e2e/cucumber/features/ocm/ocm.feature (hunk 2 above)
Comment (amrita-shrestha) [reply tests/e2e/cucumber/features/ocm/ocm.feature:3]: so you want me to write steps without step definition

File: tests/e2e/cucumber/environment/index.ts (hunk 5 above)
Comment (saw-jan) [reply tests/e2e/cucumber/environment/index.ts:95]: same like before hook using tag
```
if (tags.includes('@ocm')) {
```

Comment (amrita-shrestha) [reply tests/e2e/cucumber/environment/index.ts:95]: sorry but deleteUser function need config  **config.federatedServer = true** to be set

File: tests/e2e/cucumber/environment/index.ts [hunk 6]
```diff
@@ -78,13 +83,20 @@ Before(async function (this: World, { pickle }: ITestCaseHookParameter) {
...
   })
+
   if (!config.basicAuth) {
     const user = this.usersEnvironment.getUser({ key: 'admin' })
     if (config.keycloak) {
       await setAccessTokenForKeycloakOcisUser(user)
       await setAccessTokenForKeycloakUser(user)
     } else {
       await setAccessAndRefreshToken(user)
+      if (await isOcm(pickle)) {
```
Comment (saw-jan) [inline tests/e2e/cucumber/environment/index.ts:94]:
```suggestion
      if (isOcm(pickle)) {
```

**Pull request merged**